### ISSUE TITLE
Optikey with touchscreen

### ISSUE THREAD

**Issue by annakirkpatrick** [OptiKey/OptiKey#852]
Has anyone successfully used Optikey with a touchscreen as the input device?

I've been exploring different input devices and recently tried using Optikey on a large touchscreen monitor (think SmartBoard). I set the pointing source to mouse position and the key selection source to mouse button (left click). 

When I touched the first key, let's call it A, Optikey behaved as expected. Then, when I touched the second key, let's call it B, OptiKey reacted as if I had pressed A again. And then, after sending the keystroke for A, the orange highlight moved to key B. In fact, if, after pressing A,  I tapped anywhere on the screen (even on another window), OptiKey reacted as if I had pressed A.

Hopefully that description makes sense.  Let me know if it doesn't.

I have a good guess about what is happening here. OptiKey is registering the mouse button event before it is processing the fact that the mouse has moved. (And of course the mouse position has jumped in a discontinuous way from one key to another.) 

Obviously, this isn't expected behavior.  But, if touchscreen use is out-of-scope for this project, I get that, too.

**Comment by kmcnaught** [comment]
Hi @annakirkpatrick,

As far as I'm aware, Optikey does not support touch input but we would like it to. It's been on our wishlist since 2014 (!), and @idanrichman recently shared a potential implementation at https://github.com/OptiKey/OptiKey/issues/23#issuecomment-1174417456

Perhaps you would like to give their code a go and report back to us? 

FWIW I get the same behaviour you do with the mouse click occurring before the mouse location has changed.

**Comment by annakirkpatrick** [comment]
Thanks @kmcnaught. It's good to know what the status is, and I will take a look at the linked commit.

For what it's worth, I spent some time a couple days ago looking at other on-screen keyboard options (looking for something that would work with touch input), and nothing I found came anywhere close to matching the feature set of Optikey. In particular, the ability to have dynamic keyboards with "macro keys" that do more than just send the keystroke seems to be unique to Optikey, at least based on my cursory web search. So I would say that implementing touch support adds value not just to Optikey but also to accessibility options more broadly.
